HungryDevelopers.playground/Contents.swift

```diff
@@ -0,0 +1,119 @@
+import UIKit
+
+class Spoon {
+    var index: Int = 0
+    private let lock = NSLock()
+    
+    func pickUp() {
+        lock.lock()
+    }
+    
+    func putDown() {
+        lock.unlock()
+    }
+}
+
+class Developer {
+    var name: String?
+    var leftSpoon: Spoon?
+    var rightSpoon: Spoon?
```

Comment:
I would avoid all of these being optional. The less you can use optionals in code, the better, imo. If there were a case where a developer wouldn't have a name or wouldn't have either the right or left spoons, that would make sense to have them be optional.